File: java/P401.java
class Solution {
    public List<String> readBinaryWatch(int num) {
        List<String> answer = new ArrayList<>();
        for (int t = 0; t < 12; t++) {
            for (int m = 0; m < 60; m++) {
                if (Integer.bitCount(t*64+m) == num) {
                   answer.add(String.format("%d:%02d", t, m)); 
                }
            }
        }
        return answer;
    }
}

File: python/P401.py
class Solution(object):
    def readBinaryWatch(self, num):
        answer = []
        for x in range(1024):
            if bin(x).count('1') == num:
                h = x>>6
                m = x&0x3f
                if h <12 and m <60:
                    answer.append('%d:%02d' % (h,m))
        return answer